diff --git a/node_modules/@midnight-ntwrk/compact-js/dist/esm/effect/ContractExecutable.js b/node_modules/@midnight-ntwrk/compact-js/dist/esm/effect/ContractExecutable.js
index bc91fbf..327ca58 100644
--- a/node_modules/@midnight-ntwrk/compact-js/dist/esm/effect/ContractExecutable.js
+++ b/node_modules/@midnight-ntwrk/compact-js/dist/esm/effect/ContractExecutable.js
@@ -108,8 +108,8 @@ class ContractExecutableImpl {
         return Effect.all({
             keyConfig: Configuration.Keys,
             contract: this.createContract()
-        }).pipe(Effect.flatMap(({ keyConfig, contract }) => Effect.try({
-            try: () => {
+        }).pipe(Effect.flatMap(({ keyConfig, contract }) => Effect.tryPromise({
+            try: async () => {
                 const circuit = contract.provableCircuits[provableCircuitId];
                 if (!circuit) {
                     throw new Error(`Circuit ${this.compiledContract.tag}#${provableCircuitId} could not be found.`);
@@ -120,7 +120,7 @@ class ContractExecutableImpl {
                 const runtimeContext = createCircuitContext(circuitContext.address, zswapLocalState, circuitContext.contractState, circuitContext.privateState);
                 const initialTxContext = runtimeContext.currentQueryContext;
                 return {
-                    ...circuit(runtimeContext, ...args),
+                    ...await circuit(runtimeContext, ...args),
                     initialTxContext
                 };
             },
